src/iterator.rs

```diff
@@ -11,6 +11,7 @@ use super::*;
 use std::iter;
 
 /// Iterator over a flat-tree.
+#[derive(Debug)]
```

Comment:
What does this do?

Reply:
It allows us to do ` println!("{:?}", value)` on the struct. Unless we do embedded programming, all structs should generally have `Debug` on them!

Reply:
What happens if it's not used? Compile error?

Reply:
you can't print the struct to stdout. Also can't print any structs that use this struct to stdout. You really want to be able to when debugging, haha.